preprocessing/nextclade/src/loculus_preprocessing/prepro.py

```diff
@@ -247,6 +247,116 @@ def run_sort(
     return alerts
 
 
+def assign_segment(
+    input_unaligned_sequences: dict[str, NucleotideSequence | None],
+    unaligned_nucleotide_sequences: dict[SegmentName, NucleotideSequence | None],
+    errors: list[ProcessingAnnotation],
+    aligned_nucleotide_sequences: dict[SegmentName, NucleotideSequence | None],
+    config: Config,
+):
+    valid_segments = set()
+    duplicate_segments = set()
+    if not config.multi_segment:
+        aligned_nucleotide_sequences["main"] = None
+        if len(input_unaligned_sequences) > 1:
+            errors.append(
+                ProcessingAnnotation(
+                    unprocessedFields=[
+                        AnnotationSource(
+                            name="alignment",
+                            type=AnnotationSourceType.NUCLEOTIDE_SEQUENCE,
+                        ),
+                    ],
+                    processedFields=[
+                        AnnotationSource(
+                            name="alignment",
+                            type=AnnotationSourceType.NUCLEOTIDE_SEQUENCE,
+                        ),
+                    ],
+                    message=(
+                        f"Multiple sequences: {list(input_unaligned_sequences.keys())} found in the"
+                        f" input data, but organism: {config.organism} is single-segmented. "
+                        "Please check that your metadata and sequences are annotated correctly."
+                        "Each metadata entry should have a single corresponding fasta sequence "
+                        "entry with the same submissionId."
+                    ),
+                )
+            )
+        else:
+            _, value = next(iter(input_unaligned_sequences.items()))
+            unaligned_nucleotide_sequences["main"] = value
+        return (
+            unaligned_nucleotide_sequences,
+            aligned_nucleotide_sequences,
+            errors,
+        )
+    for segment in config.nucleotideSequences:
+        aligned_nucleotide_sequences[segment] = None
+        unaligned_segment = [
```

Comment:
update some tests that show that the dictionary can include the full submissionId and not just the segment